palette/palette.go

```diff
@@ -32,54 +39,54 @@ type DivergingPalette interface {
 	CriticalIndex() (low, high int)
 }
 
-// A ColorMap maps scalar values to colors.
-type ColorMap interface {
-	// At returns the color associated with the given value.
-	// If the value is not between Max() and Min(), an error is returned.
-	At(float64) (color.Color, error)
-
-	// Max returns the current maximum value of the ColorMap.
-	Max() float64
-
-	// SetMax sets the maximum value of the ColorMap.
-	SetMax(float64)
-
-	// Min returns the current minimum value of the ColorMap.
-	Min() float64
+// A ColorMapInt maps an integer category value to a color.
+// If there is no mapped color for the given category, an error is returned.
+type ColorMapInt interface {
+	// At returns the color associated with category cat.
+	At(cat int) (color.Color, error)
+}
 
-	// SetMin sets the minimum value of the ColorMap.
-	SetMin(float64)
+// IntMap fulfils the ColorMapInt interface, mapping integer
+// categories to colors.
+type IntMap struct {
+	Categories []int
+	Colors     []color.Color
+}
 
-	// Alpha returns the opacity value of the ColorMap.
-	Alpha() float64
+// At fulfils the ColorMapInt interface.
+func (im *IntMap) At(cat int) (color.Color, error) {
+	if len(im.Categories) != len(im.Colors) {
+		panic(fmt.Errorf("palette: number of categories (%d) != number of colors (%d)", len(im.Categories), len(im.Colors)))
+	}
+	if i := sort.SearchInts(im.Categories, cat); i < len(im.Categories) && im.Categories[i] == cat {
```

Comment:
It's conceivable that people would use a large number of categories, so I've left this as binary search, but that also wouldn't be a best practice so it might be better to use the non-binary search.